### PR TITLE
Zotero fixes for 4.1.0rc 

### PR DESCRIPTION
**Pull request by mrow4a**
Fixes: 
- https://github.com/owncloud/richdocuments/issues/527
- https://github.com/owncloud/richdocuments/issues/528

Screenshot for new input handling
Screenshot 2023-11-09 at 15 46 41

Screenshot 2023-11-10 at 08 18 12

### CODE REVIEW

**Review by jnweiger — CHANGES_REQUESTED** [review]
#528 is fixed here. Thanks.
But I am not 100% happy with the fix for #527

There is no message, when no key was entered, but there is a message when a correct key was entered. See https://github.com/owncloud/richdocuments/issues/527#issuecomment-1804266520

**Comment by mrow4a** [comment]
@jnweiger added note that 22.05.10 is required in admin settings

**Comment by jvillafanez** [comment]
> Update the existing key or save empty to delete

Just checking, is it true? Maybe I'm wrong, but if the value is empty, I think it will keep the current value or set an empty string (which I don't know how it will behave)

> There is no message, when no key was entered, but there is a message when a correct key was entered.

I agree there should be some message. I think it's fixed, but I haven't tested it.


Code looks good to me.

**Comment by mrow4a** [comment]
@jnweiger could you rereview ? the adjustments are there.

**Review by jnweiger — CHANGES_REQUESTED** [review]
Found the missing bit to fix the annoying "Zotero API key is incorrect" message.
Suggestion added.

Everthing else save button wit copy paste, etc, works fine now!

File: lib/Controller/WopiController.php [hunk 1]
```diff
@@ -232,11 +232,11 @@ public function wopiCheckFileInfo(string $documentId): JSONResponse {
...
 		$supportsLocks = $canWrite && $storage->instanceOfStorage(IPersistentLockingStorage::class);
 
-		// check
+		// user private info
+		$userPrivateInfo = [];
 		if ($res['editor'] !== '') {
 			$zoteroAPIPrivateKey = $this->appConfig->getUserValue($res['editor'], 'zoteroAPIPrivateKey');
-		} else {
-			$zoteroAPIPrivateKey = null;
+			$userPrivateInfo['ZoteroAPIKey'] = $zoteroAPIPrivateKey;
```
[the remark this answers is not in the document]
Comment (mrow4a) [reply lib/Controller/WopiController.php:239]: ok! now I finally know where your problem resided originaly! Ahhh good catch!

[the remark this answers is not in the document]
Comment (mrow4a) [reply lib/Controller/WopiController.php:239]: I will fix it otherwise

**Comment by mrow4a** [comment]
@jnweiger fixed your very good catch with empty key

**Review by jnweiger — APPROVED** [review]
Looks like we finally solved all the issues here !

Thanks @mrow4a !

**Pull request merged**